superset/migrations/shared/security_converge.py

```diff
@@ -0,0 +1,260 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from typing import Dict, List, Tuple
+
+from sqlalchemy import (
+    Column,
+    ForeignKey,
+    Integer,
+    Sequence,
+    String,
+    Table,
+    UniqueConstraint,
+)
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import Load, relationship, Session
+
+Base = declarative_base()
+
+PvmType = Tuple[str, str]
+PvmMigrationMapType = Dict[PvmType, Tuple[PvmType, ...]]
```

Comment:
Why the tuple? Will these ever be anything other than 1:1?

Reply:
They will, on downgrade we need `1:N` because I'm "recovering" for example: "can_read" -> ("can_show", "can_list", ...)